cli/src/test/java/com/expediagroup/streamplatform/streamregistry/cli/command/ApplyTest.java

```diff
@@ -0,0 +1,150 @@
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.SchemaKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ZoneKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.SpecificationEvent;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Specification;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Tag;
+
+import picocli.CommandLine;
+import picocli.CommandLine.ParseResult;
+
+public class ApplyTest {
+  private final CommandLine underTest = new CommandLine(new Apply());
+
+  private final String[] graphqlOptions = {"--streamRegistryUrl=streamRegistryUrl"};
+  private final String[] configurationOptions = {"--description=description", "--tag=a:b", "--tag=c:d", "--type=type", "--configuration={\"e\":\"f\"}"};
+
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final ObjectNode configuration = mapper.createObjectNode()
+      .put("e", "f");
+
+  private final DomainKey domainKey = new DomainKey("domain");
+  private final SchemaKey schemaKey = new SchemaKey(domainKey, "schema");
+  private final StreamKey streamKey = new StreamKey(domainKey, "stream", 1);
+  private final ZoneKey zoneKey = new ZoneKey("zone");
+  private final InfrastructureKey infrastructureKey = new InfrastructureKey(zoneKey, "infrastructure");
+  private final ProducerKey producerKey = new ProducerKey(streamKey, zoneKey, "producer");
+  private final ConsumerKey consumerKey = new ConsumerKey(streamKey, zoneKey, "consumer");
+  private final StreamBindingKey streamBindingKey = new StreamBindingKey(streamKey, infrastructureKey);
+  private final ProducerBindingKey producerBindingKey = new ProducerBindingKey(producerKey, streamBindingKey);
+  private final ConsumerBindingKey consumerBindingKey = new ConsumerBindingKey(consumerKey, streamBindingKey);
+
+  @Test
+  public void domain() {
+    ParseResult result = parseArgs("domain",
+        "--domain=domain");
+    assertEvent(result, domainKey);
+  }
+
+  @Test
+  public void schema() {
+    ParseResult result = parseArgs("schema",
+        "--domain=domain", "--schema=schema");
+    assertEvent(result, schemaKey);
+  }
+
+  @Test
+  public void stream() {
+    ParseResult result = parseArgs("stream",
+        "--domain=domain", "--stream=stream", "--version=1",
+        "--schemaDomain=schemaDomain", "--schema=schema");
+    assertEvent(result, streamKey);
+  }
+
+  @Test
+  public void zone() {
+    ParseResult result = underTest.parseArgs("zone",
+        "--streamRegistryUrl=streamRegistryUrl",
+        "--description=description", "--tag=a:b", "--tag=c:d", "--type=type", "--configuration={\"e\":\"f\"}",
```

Comment:
Consider allowing for --configurationFile="name.json" since the config could be quite large.
Equally, if this CLI tool is intended for use by other tools rather than humans perhaps it doesn't matter.

Reply:
Yes agreed. We can iterate, we just need to get something out. Would be nice to support multiple at once as well (incl from stdin).